pkg/kubernetes/overrides.go

```diff
@@ -44,6 +44,18 @@ func PatchPodSpecWithOverride(spec, override *corev1.PodSpec) (*corev1.PodSpec,
 		return nil, fmt.Errorf("can't marshal pod spec override: %w", err)
 	}
 
+	// If any of the override containers have livenessProbe overrides, the entire object should be replaced with the override.
```

Comment:
More specifically, should we check and replace `probeHandler`  instead of the whole probe?

Reply:
Replacing the entire probe seemed cleaner (instead of looking for individual probeHandlers). But, I am open to changing it if that causes less confusion.